src/simple/room/Participant.js

```diff
@@ -142,40 +187,59 @@ class Participant extends Emitter {
       audioMuted: this.audioMuted,
       videoMuted: this.videoMuted,
       screensharing: this.screensharing,
+      isTalking: this.isTalking,
       extra: this.extra,
     };
   }
 
-  updateStatus(status: Object) {
-    if (status.audioMuted !== this.audioMuted) {
+  updateStatus(status: Object, broadcast: boolean = true) {
+    let updated: boolean = false;
+    if (status.audioMuted !== undefined && status.audioMuted !== this.audioMuted) {
       if (status.audioMuted) {
-        this.onAudioMuted();
+        this.onAudioMuted(broadcast);
       } else {
-        this.onAudioUnMuted();
+        this.onAudioUnMuted(broadcast);
       }
+      updated = true;
     }
 
-    if (status.videoMuted !== this.videoMuted) {
+    if (status.videoMuted !== undefined && status.videoMuted !== this.videoMuted) {
       if (status.videoMuted) {
-        this.onVideoMuted();
+        this.onVideoMuted(broadcast);
       } else {
-        this.onVideoUnMuted();
+        this.onVideoUnMuted(broadcast);
       }
+      updated = true;
     }
 
-    if (status.screensharing !== this.screensharing) {
+    if (status.screensharing !== undefined && status.screensharing !== this.screensharing) {
       if (status.screensharing) {
-        this.onScreensharing();
+        this.onScreensharing(broadcast);
       } else {
-        this.onStopScreensharing();
+        this.onStopScreensharing(broadcast);
       }
+      updated = true;
     }
 
-    // Poor man's object comparision
-    if (JSON.stringify(this.extra) !== JSON.stringify(status.extra)) {
+    // Poor man's object comparison
+    if (status.extra !== undefined && JSON.stringify(this.extra) !== JSON.stringify(status.extra)) {
       this.extra = { ...this.extra, ...status.extra };
-      this.eventEmitter.emit(this.ON_UPDATED);
+      this.triggerUpdate(this.ON_EXTRA_CHANGE, broadcast);
+      updated = true;
     }
+
+    if (!updated) {
```

Comment:
Weird to see a `ON_UPDATED` event when `!updated`

Reply:
yeaaaah... i knew that one would make you wince -- i wasn't too sure about this one either. At one point in my refresh mayhem, i had to trigger an event even if values weren't update; i'll double check.